editoast/src/modelsv2/rolling_stock_model.rs

```diff
@@ -0,0 +1,150 @@
+use actix_web::web::Data;
+use diesel::{ExpressionMethods, QueryDsl, SelectableHelper};
+use diesel_async::RunQueryDsl;
+use editoast_derive::ModelV2;
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use utoipa::ToSchema;
+use validator::{Validate, ValidationError};
+
+use crate::error::Result;
+use crate::models::rolling_stock::rolling_stock_livery::RollingStockLiveryMetadata;
+use crate::modelsv2::*;
+use crate::schema::rolling_stock::{
+    EffortCurves, EnergySource, Gamma, RollingResistance, RollingStock, RollingStockMetadata,
+    RollingStockWithLiveries,
+};
+use crate::DbPool;
+
+use super::Changeset;
+
+#[derive(Clone, Debug, PartialEq, Deserialize, Serialize, ToSchema)]
+pub struct RollingStockSupportedSignalingSystem(pub Vec<String>);
+
+impl From<Vec<Option<String>>> for RollingStockSupportedSignalingSystem {
+    fn from(features: Vec<Option<String>>) -> Self {
+        Self(features.into_iter().flatten().collect())
+    }
+}
+impl From<RollingStockSupportedSignalingSystem> for Vec<Option<String>> {
+    fn from(features: RollingStockSupportedSignalingSystem) -> Self {
+        features.0.into_iter().map(Some).collect()
+    }
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize, ModelV2, Validate, ToSchema)]
+#[model(table = crate::tables::rolling_stock)]
+#[model(changeset(derive(Deserialize, Serialize, Debug, PartialEq), public))]
```

Comment:
```suggestion
#[model(changeset(derive(Deserialize, Serialize, Debug, PartialEq)))]
```

if possible, I'd prefer to avoid exposing the changeset's internals